strawberryfields/compilers/gaussian_merge.py

```diff
@@ -0,0 +1,376 @@
+
+    **Example:**
+
+    Consider the following Strawberry Fields program, compiled using the `'gaussian_merge'` compiler:
+
+    .. code-block:: python3
+
+        from strawberryfields.ops import Xgate, Zgate, BSgate, Rgate
+        import strawberryfields as sf
+
+        circuit = sf.Program(1)
+        with circuit.context as q:
+            Rgate(0.6)|q[0]
+            Rgate(0.2)|q[0]
+            Kgate(0.4)|q[0]
+            Rgate(0.1)|q[0]
+            Rgate(0.6)|q[0]
+            Dgate(0.01)|(q[0])
+
+        compiled_circuit = circuit.compile(compiler="gaussian_merge")
+
+    We can now print the compiled circuit, which has merged adjacent Gaussian
+    operations into singular :class:`~.GaussianTransform` operations:
+
+    >>> compiled_circuit.print()
+    GaussianTransform([[ 0.6967 -0.7174]
+      [0.7174  0.6967]]) | (q[0])
+    Kgate(0.4)|q[0]
+    GaussianTransform([[ 0.7648 -0.6442]
+      [0.6442  0.7648]]) | (q[0])
+    Dgate(0.01, 0) | (q[0])
+    """
+
+    short_name = "gaussian_merge"
+    primitives = {
+        # meta operations
+        "All",
+        "_New_modes",
+        "_Delete",
+        # state preparations
+        "Ket",
+        # measurements
+        "MeasureFock",
+        "MeasureHomodyne",
+        # single mode gates
+        "Dgate",
+        "Sgate",
+        "Rgate",
+        "Vgate",
+        "Kgate",
+        # two mode gates
+        "MZgate",
+        "sMZgate",
+        "BSgate",
+        "CKgate",
+        "S2gate",
+        "Interferometer",
+        "GaussianTransform",
+    }
+
+    decompositions = {
+        "GraphEmbed": {},
+        "BipartiteGraphEmbed": {},
+        "Gaussian": {},
+        "Pgate": {},
+        "CXgate": {},
+        "CZgate": {},
+        "Xgate": {},
+        "Zgate": {},
+        "Fouriergate": {},
+    }
+
+    prep_states = ["Ket"]
+
+    gaussian_ops = [
+        "Dgate",
+        "BSgate",
+        "S2gate",
+        "Sgate",
+        "GaussianTransform",
+        "Rgate",
+        "Interferometer",
+        "MZgate",
+        "sMZgate",
+    ]
+
+    def __init__(self):
+        self.curr_seq = None
+        self.DAG = None
+        self.new_DAG = None
+
+    def compile(self, seq, registers):
+        """Attempt to merge Gaussian operations into Gaussian Transforms in a hybrid program.
+
+        Args:
+            seq (Sequence[Command]): quantum circuit to modify
+            registers (Sequence[RegRefs]): quantum registers
+        Returns:
+            List[Command]: modified circuit
+        """
+        self.curr_seq = seq
+        while self.merge_a_gaussian_op(registers):
+            continue
+        return self.curr_seq
+
+    def merge_a_gaussian_op(self, registers):
+        """
+        Main function to merge a gaussian operation with its gaussian neighbours.
+        If merge is achieved, the method updates self.curr_seq and returns True
+        Else (merge cannot be achieved), the method returns false
```

Comment:
```suggestion
        If a merge is succesful, the method updates self.curr_seq and returns True.
        Otherwise (a merge cannot be achieved), the method returns False.
```